processes/release-process.md

```diff
@@ -123,3 +123,121 @@ versioned releases are available:
 - [{{repo}}](https://github.com/{{project.name}}/{{repo}})
 	{% endfor %}
 {% endfor %}
+
+## Release procedure
+
+This section describes how a release is done. Most of the points below are
+applicable to the wider public except for those where the actual release work
+is carried out. Note that this procedure is based on the Trustworthy Systems
+team-internal Bamboo continuous infrastructure and GitHub setup. This procedure
+is likely to change in the coming future as the Trustworthy Systems team
+migrates away from the Bamboo setup and perhaps experiments with a more
+frequent release cycle.
+
+### Release window and feeze
+
+When a release is planned, decide on the timeframe of the release window in
+which changes can still be made, and then make an announcement to the relevant
+channels that a release is coming up and let people know of the release window
+that you have decided. This is to give advance notice to contributors so that
+they can prepare for the release and get features merged in before the release
+is done. A decent timeframe for the release window is a month.
+
+The relevant channels for the announcement are:
+    - [seL4 discourse](https://sel4.discourse.group/)
+    - seL4 announcement mailing list: announce@sel4.systems
+    - [seL4 Mattermost](https://mattermost.trustworthy.systems/sel4-external/)
+
+Upon the end of the release window, announce the close of the window and the
+start of the release freeze on the relevant channels. Pull requests should not
+be merged during the release freeze except for those that fix issues and bugs
+that occurred during the release window. Work towards getting the continuous
+integration builds to be green while working on the next step.
+
+### Preparing changelogs
+
+Before preparing the changelogs for each project, review the changes of each
+project since the last release and decide a new version number as according to
+the [Version Numbers](#version-numbers) section.
+
+Clone the release tool from GitHub (you may need to ask for permission).
+
+Move to the directory of the release tool that you cloned and execute the following while replacing the fields:
```

Comment:
We should probably mention the `seL4_repo` here (and say that it is currently private). It'd be nice to make this public at some point, but not yet.

Reply:
Yep.